src/programacion/empresaprogramacion/Empleado.java

```diff
@@ -1,8 +1,79 @@
 package programacion.empresaprogramacion;
+import java.util.Comparator;
+
+public abstract class Empleado implements Comparable<Empleado> {
+  private String idEmpleado;
+  private String nombre;
+  private double sueldo;
+  private TipoEmpleado tipo;
+  private final static String idEmpleadoRegexp = "[A-Z]{1}[8]{1}[0-9]{3}";
```

Comment:
Recuérdamen que revisemos esta regex en clase. Si quieres que comience por una letra mayúscula, luego un 8 y luego 3 dígitos, se podría abreviar así:
```suggestion
  private final static String idEmpleadoRegexp = "[A-Z]8[0-9]{3}";
```